### PR TITLE
Return error instead of bool

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: seeker_test.go [hunk 1]
```diff
@@ -26,15 +26,16 @@ func TestSeeker(t *testing.T) {
...
-	s := NewSeeker(f)
+	s := New(f)
 
 	if err = s.SeekToStart(); err != nil {
 		t.Fatal(err)
 	}
 
 	var str string
-	if err = s.ReadLine(func(buf *bytes.Buffer) {
+	if err = s.ReadLine(func(buf *bytes.Buffer) error {
```
Comment (shahzilabid) [inline seeker_test.go:36]: Hmm all the funcs you pass into ReadLine never return an err..always nil..is it needed?

Comment (itsmontoya) [reply seeker_test.go:36]: We do in mrT, see this line for seeker:
https://github.com/itsmontoya/seeker/pull/1/files/474907a3be1798e5ac976de8ee02e8e61c1942f0#diff-fddc5bba3483c0030216dceff265ae46R267

Comment (itsmontoya) [reply seeker_test.go:36]: So the func passed to ReadLines (https://github.com/itsmontoya/seeker/blob/474907a3be1798e5ac976de8ee02e8e61c1942f0/seeker.go#L277) can end the iteration early

When an error is returned from the provided fn, that error will pass through ReadLine or ReadLines so the calling func is aware.

Comment (shahzilabid) [reply seeker_test.go:36]: Gotcha

**Review by shahzilabid — APPROVED** [review]

**Pull request merged**